modules/bridge/src/main/ts/ephox/bridge/api/Registry.ts

```diff
@@ -49,7 +49,7 @@ export const create = (): Registry => {
   const contextMenus: Record<string, ContextMenuApi> = {};
   const contextToolbars: Record<string, ContextToolbarApi | ContextFormApi> = {};
   const sidebars: Record<string, SidebarApi> = {};
-  const add = (collection, type: string) => (name: string, spec: any): void => collection[name.toLowerCase()] = { ...spec, type };
+  const add = (collection, type: string) => (name: string, spec: any): void => collection[name.toLowerCase()] = { ...(() => {if (spec.icon !== undefined) {spec.icon = spec.icon.toLowerCase();};return spec;})(), type };
```

Comment:
This isn't a great place to be doing this, as it's manipulating the spec. This should probably be fixed in `modules/tinymce/src/themes/silver/main/ts/ui/icons/Icons.ts` instead.

A couple other things just for future reference:
- This really needs some formatting (we seem to have lost this rule when moving to eslint as well)
- You're mutating the spec, which is something we avoid at all costs. If you need to edit this, clone it and make a new object instead.

Reply:
Also, this should really have at least one test added to make sure we don't accidentally break this in future.

Reply:
I thought silver was just our canonical theme? It is a better place to add the fix but wouldn't that put the responsibility on anyone who writes their own theme to do the same?

Reply:
It is, but it's the part where the bug really lies. As there are many other icons that are used that don't go via specs, so it cannot be done in the registry.

Reply:
Ideally, that icon lookup code should really be in core (registry perhaps) rather than the theme, to similar.

Reply:
Per the discussion in Slack, I'll revamp this PR to be a quick fix to the silver theme and open a new issue for a bigger/longer-term fix that we can triage later.

Reply:
I've added the quickfix to all of the icon retrieval methods and added a test to `IconsTest`. I had to alter `AlertBannerTest` because its mock icon collection used a mixed case. Of course, it isn't (and hasn't) been possible to add mixed case icon names to the registry but `AlertBannerTest` manually constructed its collection for testing purposes.